src/accumulator/accumulated_map.rs

```diff
@@ -74,6 +74,14 @@ impl InputAccumulatedValues {
     pub const fn is_empty(self) -> bool {
         matches!(self, Self::Empty)
     }
+
+    pub fn or_shortcircuiting(self, other: impl FnOnce() -> Self) -> Self {
```

Comment:
Would `or` be enough? Similar to `Result::or` and `Option::or` that both "short circuit" if `self` is `Ok` or `Some`